onnxruntime/core/optimizer/matmul_add_fusion.cc

```diff
@@ -15,6 +15,9 @@ Status MatMulAddFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
   GraphViewer graph_viewer(graph);
   const auto& node_topology_list = graph_viewer.GetNodesInTopologicalOrder();
 
+  std::unordered_set<const Node*> attn_ln_nodes;
+  std::unordered_set<const Node*> attn_add_nodes;
```

Comment:
This two sets are used to remember the attention sub-graph (which has 4 MatMul+Add pair) so that when it hits another pair on the all-node FOR loop, we can skip quickly. Just added a comment there to explain.